src/YagrCore/plugins/tooltip/utils.ts

```diff
@@ -8,3 +10,65 @@ export function escapeHTML(html: string) {
     elem.innerText = html;
     return elem.innerHTML;
 }
+
+type FindClosestAreaOptions = {
+    /** cursor X */
+    x: number;
+    /** cursor Y */
+    y: number;
+    series: AlignedData;
+};
+
+export const findClosestLines = ({x, y, series}: FindClosestAreaOptions) => {
+    const [timeline, ...areas] = series as number[][];
+
+    let i1 = 0,
+        i2 = 0;
+
+    // finding current timeline
+    timeline.forEach((t, i) => {
+        if (i1 || i2) {
+            return;
+        }
+        if (t === x) {
+            i1 = i2 = i;
+            return;
+        }
+        if (t > x) {
+            i1 = i - 1;
+            i2 = i;
+        }
+    });
+
+    let minLower = Number.MAX_VALUE,
+        minHigher = Number.MAX_VALUE;
+    let iLower: number | null = null,
+        iHigher: number | null = null;
+
+    const x1 = timeline[i1];
+    const x2 = timeline[i2];
+
+    areas
```

Comment:
Let's try to reduce number of possible iterations. I understand that in most cases it doesn't do any issues, but I'd like to keep calculations as fast as possible.